src/devices/cpu/hcd62121/hcd62121.cpp

```diff
@@ -1152,6 +1213,87 @@ void hcd62121_cpu_device::execute_run()
 			}
 			break;
 
+		case 0x40:      /* shrb/shlb ir1,8 */
+		case 0x41:      /* shrw/shlw ir1,8 */
+		case 0x42:      /* shrq/shlq ir1,8 */
+		case 0x43:      /* shrt/shlt ir1,8 */
+			{
+				int size = datasize(op);
+				u8 reg1 = read_op();
+
+				// TODO - read_ireg should support reading this
+				if (reg1 & 0x80) {
+					read_ireg(size, (reg1 & 0x7f) - size + 1);
+					for (int i=0; i < size - 1; i++) {
+						m_temp1[i] = m_temp1[i + 1];
+					}
+					m_temp1[size-1] = 0;
+					write_ireg(size, (reg1 & 0x7f) - size + 1);
+				}
+				else
+				{
+					read_ireg(size, reg1);
+					for (int i = size-1; i > 0; i--) {
+						m_temp1[i] = m_temp1[i - 1];
+					}
+					m_temp1[0] = 0;
+					write_ireg(size, reg1);
+				}
```

Comment:
Same here, please use Allman formatting for consistency with the rest of the file.